### PR TITLE
feat: workflow usage

### PR DESCRIPTION
**Pull request by PoorlyDefinedBehaviour**
- Adds a query to fetch a summary of the workflow usage

Screenshot 2023-07-28 at 10 52 48

Part of https://github.com/grafana/hosted-grafana/issues/2556
Closes https://github.com/grafana/hosted-grafana/issues/4407
Related https://github.com/grafana/github-datasource/issues/105 (i don't know if this PR is enough to close the issue because it is hard to get the data we need from the GitHub api due to rate limits.)

### CODE REVIEW

File: pkg/github/client/client.go [hunk 1]
```diff
@@ -51,3 +83,172 @@ func (client *Client) Query(ctx context.Context, q interface{}, variables map[st
...
+	}
+
+	conclusions := map[string]uint64{
+		"success":   0,
+		"failure":   0,
+		"cancelled": 0,
+		"skipped":   0,
+	}
+
+	usageDuration := time.Second * 0
```
Comment (baldm0mma) [inline pkg/github/client/client.go:108]: Won't this always evaluate to 0? Is there a reason it's not simply set to 0?

Comment (PoorlyDefinedBehaviour) [reply pkg/github/client/client.go:108]: If i just set it to zero the type of `usageDuration` won't be `time.Duration` and then down below when we try to add it to a value of type `time.Duration`, we'll get a type error:
```go
usageDuration += duration
```

I can change it to `var usageDuration time.Duration` if you think it makes it easier to understand

Comment (baldm0mma) [reply pkg/github/client/client.go:108]: Ah! That make sense! I think `var usageDuration time.Duration` would help the readability a bunch. :)

**Review by baldm0mma — APPROVED** [review]
Thanks for the change. LGTM! I'd love to have one more of us from the squad take a look, just for coverage, so I'm going to repost this in our public channel.

File: pkg/github/client/client.go (hunk 1 above)
Comment (PoorlyDefinedBehaviour) [reply pkg/github/client/client.go]: Done

**Review by codeincarnate — APPROVED** [review]

**Comment by PKatBK** [comment]
when will this be implemented?

**Comment by zoltanbedi** [comment]
@PKatBK you need to update the plugin to see this. Also try to scroll down it might be there at the bottom.

**Comment by richgreen-moj** [comment]
Has anyone worked out how to add multiple workflows into one panel?
I've managed to do that using a transformation but I can't tell which workflow each row applies to...
Screenshot 2024-05-09 at 15 47 51

**Pull request merged**